hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java

```diff
@@ -369,6 +391,36 @@ public String summarizedWith(DualStateAccessor dualStateAccessor) {
 				congestionStartsDesc();
 	}
 
+	public void incrementBlockNo() {
+		blockNo++;
+	}
+
+	public long getBlockNo() {
+		return blockNo;
+	}
+
+	public void cachePreviousBlockHash(final Hash hash) {
+		//When the cache exceeds 256 entries we need to remove the first entry to free one slot for the new block
+		if(blockNumberToHash.size() > 255 && !blockNumberToHash.containsKey(blockNo)) {
```

Comment:
With the suggested above changes, this would become:
```suggestion
void finishCurrentBlock(final Hash hash, final Instant firstNewBlockTime) {
    if (blockHashes.size() > 255) {
        blockHashes.remove(blockHashes.firstKey());
    }
    blockHashes.put(blockNo, hash);
    blockNo++;
    firstNewBlockTime = null; 		     		    
```